### PR TITLE
539 add apply to property

### PR DESCRIPTION
**Pull request by ProkopovVitaliy**
**Description** (*)

Added possibility set `apply_to` property for product  eav attribute. 

There are two variants how to use this property in data patch generation:

1. Apply attribute for all available product types in the system:
Screenshot 2021-04-12 at 13 53 43
If  `Apply to all products types` selected no specific product type will be specified for the attribute.

Screenshot 2021-04-12 at 13 54 15

2. Specify product types:

Unselect `Apply to all products types`, and it will be possible to specify product types:
Screenshot 2021-04-12 at 13 55 11

Screenshot 2021-04-12 at 13 55 29

After that will be generated data patch with specified product types:

Screenshot 2021-04-12 at 13 55 44

**Fixed Issues (if relevant)**

1. Fixes magento/magento2-phpstorm-plugin#539

**Questions or comments**

**Contribution checklist** (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with integration/functional tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by eduard13 — COMMENTED** [review]
Hi @ProkopovVitaliy, thank you for the provided changes. But I guess providing a multiple select with available product types, would be much more user friendly. Additionally, we'll avoid entering some typos within that field.

Please let me know your thoughts.

Thank you.

**Comment by ProkopovVitaliy** [comment]
> Hi @ProkopovVitaliy, thank you for the provided changes. But I guess providing a multiple select with available product types, would be much more user friendly. Additionally, we'll avoid entering some typos within that field.
> 
> Please let me know your thoughts.
> 
> Thank you.

Hi @eduard13 Yes you are right, this is more user friendly. But if developer have custom product type in their store, fore example `sellable`, we cannot predict such options and it can be some restriction for interact with our dialog window.So that why i added "help text block" which contain default types:
`Enter comma separated product types for which attribute will available, for example: simple,configurable,bundle,virtual,downloadable,category. Or leave this property blank if you want apply attribute for all product types.`

What about your opinion ?

**Comment by DmitryFurs** [comment]
@ProkopovVitaliy @eduard13 Magento provides **product_types.xml** file that describes the product types. Is it possible to implement support for this file for the **apply_to** option? When typing manually, there is a chance of making a typo. There may also be third-party product types. If there was multiselect option, it would be very convenient. Thanks!

https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Downloadable/etc/product_types.xml#L9
https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/ConfigurableProduct/etc/product_types.xml#L9

**Comment by VitaliyBoyko** [comment]
@ProkopovVitaliy 
So we can add a file indexer for `product_types.xml`, and build options accordingly.

**Comment by ProkopovVitaliy** [comment]
@eduard13 I finalized this functionality. Could you please review it?

**Main points:** 

- Added indexer for `product_types.xml`
- Added multi-select for `apply_to` property that based on parsed result form `product_types.xml` files.

**Overview:**
1. Create attribute for all product types: 
Screenshot 2021-04-13 at 14 29 38
Screenshot 2021-04-13 at 14 29 52
There are no changes.

2. Create attribute  for specified product types:
Screenshot 2021-04-13 at 14 27 33
Screenshot 2021-04-13 at 14 28 32
As we can see here added new multi-select field with product types. 

3. And the new one. When we decide  to create new product type we will see him in the multi-select: 
Create new product type:
Screenshot 2021-04-13 at 14 33 08

Run reindex for plugin:
Screenshot 2021-04-13 at 14 33 24

Open window for generation and try generate attribute for new created product type:
Screenshot 2021-04-13 at 14 35 34
Screenshot 2021-04-13 at 14 35 50

File: src/com/magento/idea/magento2plugin/actions/generation/generator/util/eav/ProductAttributeMapper.java [hunk 1]
```diff
@@ -71,6 +71,11 @@
         mappedAttributes.put(EavAttribute.VISIBLE_ON_FRONT.getAttribute(),
                 Boolean.toString(eavEntityData.isVisibleOnFront()));
 
+        if (eavEntityData.getApplyTo() != null && !eavEntityData.getApplyTo().isEmpty()) {
+            mappedAttributes.put(EavAttribute.APPLY_TO.getAttribute(),
+                    wrapStringValueForTemplate(eavEntityData.getApplyTo()));
```
Comment (eduard13) [inline src/com/magento/idea/magento2plugin/actions/generation/generator/util/eav/ProductAttributeMapper.java:76]: @ProkopovVitaliy, just want to bring your attention, that breaking a multiple arguments to a new line, should move all the arguments to a new empty line, like following.

No action is needed here, just wanted to add a remark, it also applies to PHP code.
```suggestion
            mappedAttributes.put(
                    EavAttribute.VISIBLE_ON_FRONT.getAttribute(),
                    Boolean.toString(eavEntityData.isVisibleOnFront())
            );
```

**Review by eduard13 — COMMENTED** [review]
No action is needed here, just wanted to add a remark, it also applies to PHP code.

File: src/com/magento/idea/magento2plugin/actions/generation/generator/util/eav/ProductAttributeMapper.java (hunk 1 above)
Comment (ProkopovVitaliy) [reply src/com/magento/idea/magento2plugin/actions/generation/generator/util/eav/ProductAttributeMapper.java:76]: @eduard13 Thanks for recommendation, really it's looks better in your variant. I will take this into account.

Thanks!

**Review by eduard13 — APPROVED** [review]
Thank you for the feature 👍.

**Pull request merged**